File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.java
import java.util.Arrays;
class Solution {
    public String solution(String s, String skip, int index) {
        StringBuffer answer = new StringBuffer();
        char[] s_ch = s.toCharArray();
        char[] skip_ch = skip.toCharArray();
        Arrays.sort(skip_ch);
        for(char ch : s_ch){
            for(int i=0; i<index; i++){
                ch++;
                for(char sch : skip_ch){
                    if(sch == ch) ch++;
                }
                if(ch>'z'){
                    ch = (char)(ch-26);
                    for(char sch : skip_ch){
                        if(sch == ch) ch++;
                    }
                }
            }
            if(ch>'z') ch = (char)(ch-26);
            answer.append(ch);
        }
        return answer.toString();
    }
}

File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    let skipCode = [...skip].map(e => e.charCodeAt());
    return [...s].reduce((a,c) => {
        let code = c.charCodeAt(0);
        for(var i=0; i<index; i++){
            code++;
            if(code > 122) code = 97;
            if(skipCode.includes(code)) i--;
        }
        return a+=String.fromCharCode(code);
    },'');
}